include/rocksdb/advanced_options.h

```diff
@@ -758,6 +768,13 @@ struct AdvancedColumnFamilyOptions {
   // data is left uncompressed (unless compression is also requested).
   uint64_t sample_for_compression = 0;
 
+  // EXPERIMENTAL
+  // The feature is still in development and is incomplete.
+  // If this option is set, when creating bottommost files, pass this
+  // temperature to FileSystem used. Should be no-op for default FileSystem
+  // and users need to plug in their own FileSystem to take advantage of it.
+  Temperature bottommost_temperature = Temperature::kUnknown;
```

Comment:
kUnknown is in the design doc @riversand963 wrote:)

Reply:
> What do you think about user just telling us the trend of temperature vs. data age? One of: older data is colder, older data is hotter, or unknown trend. The implementation could use it to only separate bottommost vs. non-bottommost as the first step. Maybe later it could assign more granular temperatures to levels according to the trend. Or we can refine data age to not be only about level, but instead come from seqno or timestamp in properties.

I changed my mind about this on second thought. I think the `bottommost_temperature` is better now with the caveat that temperature really refers to an abstract notion of a minimum file placement tier -- not specifically access frequency as the name implies. Abstracting the file placement tiers (hot, warm, cold) allow application to define tier-specific SLA and Env to choose any media that meets that SLA without RocksDB knowing.

RocksDB only needs to get involved when it comes to the LSM shape such as assigning the oldest data (bottommost level) to a minimum placement tier. The application can configure this based on its knowledge of query-level SLA, how placing one level in a different tier might impact that SLA, its guarantee on how much data is in bottom level (90% in default dynamic leveled), and its knowledge of how much capacity it has in the various tiers. RocksDB simply acts to evaluate the "abstract minimum file placement tier" according to its LSM knowledge and application config, and pass that to the Env -- all without knowing anything about the storage media.